jobbergate-api/tests/apps/job_submissions/test_routers.py

```diff
@@ -2079,3 +2124,498 @@ async def test_job_submissions_agent_active__returns_400_if_token_does_not_carry
+
+    mocked_session_execute = mock.AsyncMock()
+    mocked_session_execute.return_value.fetchall = mock.Mock()
+
+    base_time = int(datetime.now().timestamp())
+    raw_data = generate_job_submission_metric_columns(base_time, num_rows)
+
+    mocked_session_execute.return_value.fetchall.return_value = [
+        JobSubmissionMetricSchema.from_iterable(data_point).model_dump(exclude=["step", "task"]).values()
+        for data_point in raw_data
+    ]
+
+    if node_host is not None:
+        where_statement = "WHERE job_submission_id = :job_submission_id AND node_host = :node_host"
+        match sample_rate:
+            case JobSubmissionMetricSampleRate.ten_seconds:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_10_seconds_by_node
+            case JobSubmissionMetricSampleRate.one_minute:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_1_minute_by_node
+            case JobSubmissionMetricSampleRate.ten_minutes:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_10_minutes_by_node
+            case JobSubmissionMetricSampleRate.one_hour:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_1_hour_by_node
+            case JobSubmissionMetricSampleRate.one_week:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_1_week_by_node
+    else:
+        where_statement = "WHERE job_submission_id = :job_submission_id"
+        match sample_rate:
+            case JobSubmissionMetricSampleRate.ten_seconds:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_10_seconds_all_nodes
+            case JobSubmissionMetricSampleRate.one_minute:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_1_minute_all_nodes
+            case JobSubmissionMetricSampleRate.ten_minutes:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_10_minutes_all_nodes
+            case JobSubmissionMetricSampleRate.one_hour:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_1_hour_all_nodes
+            case JobSubmissionMetricSampleRate.one_week:
+                view_name = JobSubmissionMetricAggregateNames.metrics_nodes_mv_1_week_all_nodes
+
+    expected_sql_query = dedent(
+        f"""
+        SELECT bucket,
+            node_host,
+            cpu_frequency,
+            cpu_time,
+            cpu_utilization,
+            gpu_memory,
+            gpu_utilization,
+            page_faults,
+            memory_rss,
+            memory_virtual,
+            disk_read,
+            disk_write
+        FROM {view_name}
+        {where_statement}
+        AND bucket >= :start_time
+        AND bucket <= :end_time
+        ORDER BY bucket
+        """
+    )
+
+    start_time = datetime(2021, 1, 1, 0, 0, 0, tzinfo=timezone.utc)
+    end_time = datetime(2021, 1, 1, random_hour_interval, 0, 0, tzinfo=timezone.utc)
```

Comment:
We have been using `pendulum` in Jobbergate to make timestamps like this much more readable. I highly suggest you use it here (and probably other places you use `datetime` as well)